validations/src/validations.rs

```diff
@@ -3569,3 +3388,1162 @@ mod tests {
+        assert_eq!((p02 * 1_000_000_f64).round() as i128, 100);
+
+        // Repeat test with 100 witnesses
+        // Active identity with 1% of the reputation
+        let (t01, p01) = calculate_reppoe_threshold_v3(&rep_engine, &id_rep, 101, 2000);
+        assert_eq!(t01, Hash::with_first_u32(0xFFFF_FFFF));
+        assert_eq!((p01 * 100_f64).round() as i128, 100);
+        // Inactive identity with 0 reputation
+        let (t02, p02) = calculate_reppoe_threshold_v3(&rep_engine, &id_no_active, 101, 2000);
+        assert_eq!(t02, Hash::with_first_u32(0x00069e54));
+        assert_eq!((p02 * 1_000_000_f64).round() as i128, 101);
+    }
+
+    #[test]
+    // FIXME: Allow for now, since there is no safe cast function from a usize to float yet
+    #[allow(clippy::cast_possible_truncation)]
+    fn reppoe_worst_case() {
+        // Check the worst case reppoe probability in mainnet with an empty ARS, a data request with
+        // the maximum number of witnesses, and taking into account the extra commit rounds.
+
+        fn max_num_witnesses_for_dr_weigth(max_dr_weight: u32) -> u16 {
+            let mut dr = DRTransaction::default();
+            let mut num_witnesses = 1;
+
+            while dr.weight() < max_dr_weight {
+                dr.body.dr_output.witnesses = num_witnesses;
+                num_witnesses += 1;
+            }
+
+            num_witnesses - 1
+        }
+
+        // TODO: copied from chain.rs, make public instead?
+        fn calculate_backup_witnesses(witnesses: u16, commit_round: u16) -> u16 {
+            let exponent = u32::from(commit_round).saturating_sub(1);
+            let coefficient = 2_u16.saturating_pow(exponent);
+
+            witnesses.saturating_mul(coefficient) / 2
+        }
+
+        let consensus_constants = witnet_config::config::consensus_constants_from_partial(
+            &witnet_data_structures::chain::PartialConsensusConstants::default(),
+            &witnet_config::defaults::Mainnet,
+        );
+        let max_dr_weight = consensus_constants.max_dr_weight;
+        // Need to use extra_rounds + 1 because this variable represents the additional rounds
+        let max_commit_rounds = consensus_constants.extra_rounds + 1;
+        let minimum_difficulty = consensus_constants.minimum_difficulty;
+
+        let max_witnesses = max_num_witnesses_for_dr_weigth(max_dr_weight);
+        let max_backup_witnesses = calculate_backup_witnesses(max_witnesses, max_commit_rounds);
+
+        assert_eq!(max_witnesses, 126);
+        assert_eq!(max_witnesses + max_backup_witnesses, 630);
+
+        // Empty ARS
+        let rep_engine = ReputationEngine::new(1000);
+        let id1 = PublicKeyHash::from_bytes(&[1; 20]).unwrap();
+
+        let (_t00, p00) = calculate_reppoe_threshold_v3(
+            &rep_engine,
+            &id1,
+            max_witnesses + max_backup_witnesses,
+            minimum_difficulty,
+        );
+
+        assert_eq!((p00 * 100_f64).round() as i128, 32);
```

Comment:
This is a test that checks the worst case reppoe probability in mainnet with an empty ARS, a data request with the maximum number of witnesses allowed by the block weight limits, and taking into account the extra commit rounds.

The result is a data request with 126 witnesses (630 after adding the backup witnesses), where a node with no reputation has a 32% chance to create a valid commitment. This 32% will only be used during one epoch, if the previous commit rounds resulted in fewer commits than expected. Here are the probabilities of the other rounds for comparison:

```
32% for round 4
19% for round 3
13% for round 2
9% for round 1
```

Also, it is not possible to include other data requests while a data request with maximum weight is being resolved, so in order to achieve a "round 4" probability in every epoch, the data request would need have around 30 witnesses, to simultaneously have 4 active requests, resulting in a 7% probability in round 4 and proportionally smaller probabilities in the other rounds.